chore(api): only auto liquid probe with clean tips

ryanthecoder opened the pull request:
<!--
Thanks for taking the time to open a Pull Request (PR)! Please make sure you've read the "Opening Pull Requests" section of our Contributing Guide:

https://github.com/Opentrons/opentrons/blob/edge/CONTRIBUTING.md#opening-pull-requests

GitHub provides robust markdown to format your PR. Links, diagrams, pictures, and videos along with text formatting make it possible to create a rich and informative PR. For more information on GitHub markdown, see:

https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax

To ensure your code is reviewed quickly and thoroughly, please fill out the sections below to the best of your ability!
-->

# Overview

If a tip is "wet" it can trigger a false positive, or worse, pull up any remnant drops into the stem.

This adds a variant of the "empty" state of to the pipette state called "clean". When a tip is picked up it now marks it as "clean" instead of empty. This just sets the clean flag in the state and otherwise does the same thing as empty.

Any subsequent liquid action marks the "clean" state as false.

<!--
Describe your PR at a high level. State acceptance criteria and how this PR fits into other work. Link issues, PRs, and other relevant resources.
-->

## Test Plan and Hands on Testing

<!--
Describe your testing of the PR. Emphasize testing not reflected in the code. Attach protocols, logs, screenshots and any other assets that support your testing.
-->

## Changelog

<!--
List changes introduced by this PR considering future developers and the end user. Give careful thought and clear documentation to breaking changes.
-->

## Review requests

<!--
- What do you need from reviewers to feel confident this PR is ready to merge?
- Ask questions.
-->

## Risk assessment

<!--
- Indicate the level of attention this PR needs.
- Provide context to guide reviewers.
- Discuss trade-offs, coupling, and side effects.
- Look for the possibility, even if you think it's small, that your change may affect some other part of the system.
  - For instance, changing return tip behavior may also change the behavior of labware calibration.
- How do your unit tests and on hands on testing mitigate this PR's risks and the risk of future regressions?
- Especially in high risk PRs, explain how you know your testing is enough.
-->

ryanthecoder committed:
add way to check if tip is clean

ryanthecoder committed:
check if tip is clean before an auto-probe

ryanthecoder committed:
fix old tests

ryanthecoder committed:
add a new test

sfoster1 reviewed (changes requested):
I see the intent with clean being part of the fluid state because it is sort of "empty but also never has been other than empty", but it's really confusing to have it in the fluid state and to have all the functions that it set it be like `set_fluid_clean()`. Can we break it out into its own boolean state please? That way we can have empty and clean, or empty and not clean. 

We also probably should extend this to tracking the clean state of tips that have been returned, but I don't think we track tip return state at all in the engine so we can do that in another pr.

ryanthecoder committed:
make 'clean' a part of empty fluid instead of it's own fluid

sfoster1 reviewed (approved):
Excellent, thank you!

github-actions[bot] committed:
fix(analyses-snapshot-testing): heal EXEC-781-only-auto-probe-with-clean-tips snapshots (#17840)

This PR was requested on the PR
https://github.com/Opentrons/opentrons/pull/17808

Co-authored-by: sfoster1 <3091648+sfoster1@users.noreply.github.com>
